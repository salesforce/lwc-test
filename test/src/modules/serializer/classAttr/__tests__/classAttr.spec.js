/*
 * Copyright (c) 2018, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: MIT
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
 */
import { createElement } from 'lwc';
import ClassAttr from '../classAttr';

it('serializes component with different whitespace', () => {
    const elm = createElement('serializer-component', { is: ClassAttr });
    document.body.appendChild(elm);

    expect(elm).toMatchSnapshot();
});
