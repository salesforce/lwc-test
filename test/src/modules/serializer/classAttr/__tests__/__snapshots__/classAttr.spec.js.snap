// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serializes component with different whitespace 1`] = `
<serializer-component>
  #shadow-root(open)
    <!-- static class, static optimized -->
    <div
      class="foo bar"
    />
    <div
      class="foo bar"
    />
    <div
      class="foo bar"
    />
    <div
      class="foo bar baz"
    />
    <div />
    <div />
    <!-- static class, not static optimized -->
    <div
      class="foo bar"
    />
    <div
      class="foo bar"
    />
    <div
      class="foo bar"
    />
    <div
      class="foo bar baz"
    />
    <div />
    <div />
    <!-- dynamic class, static optimized -->
    <div />
    <div />
    <div />
    <div />
    <div
      class="foo bar"
    />
    <div
      class="foo bar"
    />
    <div
      class="foo bar"
    />
    <div
      class="foo bar"
    />
    <div
      class="foo bar baz"
    />
    <!-- dynamic class, not static optimized -->
    <div />
    <div />
    <div />
    <div />
    <div
      class="foo bar"
    />
    <div
      class="foo bar"
    />
    <div
      class="foo bar"
    />
    <div
      class="foo bar"
    />
    <div
      class="foo bar baz"
    />
</serializer-component>
`;
